streaming/src/test/scala/org/apache/spark/streaming/ReceiverSuite.scala

```diff
@@ -129,6 +130,41 @@ class ReceiverSuite extends TestSuiteBase with Timeouts with Serializable {
     }
   }
 
+  test("kill receiver") {
+    val sparkConf = new SparkConf()
+      .setMaster("local[4]")
+      .setAppName(framework)
+    val batchDuration = Milliseconds(1000)
+    withStreamingContext(new StreamingContext(sparkConf, batchDuration)) { ssc =>
+      val receiver = new FakeReceiver
+      val executor = new FakeReceiverSupervisor(receiver)
+      val task = new FakeReceiverTask(0, executor)
+
+      new Thread("Receiver Task") {
+        override def run(): Unit = {
+          task.run(0, 0)
+        }
+      }.start()
+      Thread.sleep(2000)
```

Comment:
If we can help it, we shouldn't be writing any new streaming tests that rely on Thread.sleep, since this has been a major cause of test flakiness.  See the use of ManualClock in the other streaming tests for examples of how to write tests which aren't dependent on wall-clock time.